File: dietdisplay.web/src/components/models/Meal.ts
export interface Meal {
    ingredients: Ingredient[];
    preparation: string;
    type: MealType;
}

export interface Ingredient {
    name: string;
    quantity: number;
}

export enum MealType {
    Breakfast = "Śniadanie",
    SecondBreakfast = "Drugie śniadanie",
    Dinner = "Obiad",
    Tea = "Podwieczorek",
    Supper = "Kolacja",
    Snack = "Przekąska"
}

File: DietDisplay.API/Model/Meal.cs
namespace DietDisplay.API.Model
{
    public class Meal
    {
        public required int ID { get; init; }
        public required Ingredient[] Ingredients { get; init; }
        public required MealType MealType { get; init; }
        public required string Preparation { get; init; }
    }
}
